chips/ibex/src/chip.rs

```diff
@@ -21,16 +22,18 @@ use crate::usbdev;
 
 pub const CHIP_FREQ: u32 = 50_000_000;
 
-pub struct Ibex {
+pub struct Ibex<A: 'static + Alarm<'static>> {
     userspace_kernel_boundary: SysCall,
     pmp: rv32i::pmp::PMPConfig,
+    systick: kernel::VirtualSystick<A>,
 }
 
-impl Ibex {
-    pub unsafe fn new() -> Ibex {
-        Ibex {
+impl<A: 'static + Alarm<'static>> Ibex<A> {
+    pub unsafe fn new(alarm: &'static A) -> Self {
+        Self {
             userspace_kernel_boundary: SysCall::new(),
             pmp: rv32i::pmp::PMPConfig::new(4),
+            systick: kernel::VirtualSystick::new(alarm), //TODO: Make Optional
```

Comment:
Does this really need to be optional?

Reply:
It does not, the comment has been removed.